datalad/support/dataset.py

```diff
@@ -0,0 +1,201 @@
+
+    def get_dataset_handles(self, pattern=None, fulfilled=None):
+        """Get paths to all known dataset_handles (subdatasets),
+        optionally matching a specific name pattern.
+
+        If fulfilled is True, only paths to fullfiled handles are returned,
+        if False, only paths to unfulfilled handles are returned.
+
+
+        Parameters
+        ----------
+        pattern
+        fulfilled
+
+        Returns
+        -------
+        list of str
+          (paths)
+        """
+        raise NotImplementedError("TODO")
+
+    def get_file_handles(self, pattern=None, fulfilled=None):
+        """Get paths to all known file_handles, optionally matching a specific
+        name pattern.
+
+        If fulfilled is True, only paths to fullfiled handles are returned,
+        if False, only paths to unfulfilled handles are returned.
+
+        Parameters
+        ----------
+        pattern: str
+        fulfilled: bool
+
+        Returns
+        -------
+        list of str
+          (paths)
+        """
+        raise NotImplementedError("TODO")
+
+    def record_state(self, auto_add_changes=True, message=str,
+                     update_superdataset=False, version=None):
+        """
+
+        Parameters
+        ----------
+        auto_add_changes: bool
+        message: str
+        update_superdataset: bool
+        version: str
+        """
+        raise NotImplementedError("TODO")
+
+    def set_state(self, whereto):
+        """Something that can be used to checkout a particular state
+        (tag, commit) to "undo" a change or switch to a otherwise desired
+        previous state.
+
+        Parameters
+        ----------
+        whereto: str
+        """
+        raise NotImplementedError("TODO")
+
+    def get_vcs(self):
+        """Get an instance of the version control system/repo for this dataset,
+        or None if there is none yet.
+
+        If creating an instance of GitRepo is guaranteed to be really cheap
+        this could also serve as a test whether a repo is present.
+
+        Returns
+        -------
+        GitRepo
+        """
+        if self._vcs is None:
+            try:
+                self._vcs = GitRepo(self._path, create=False)
+            except InvalidGitRepositoryError:
+                return None
+
+        return self._vcs
+
+    def is_installed(self, ensure="complete"):
+        """Returns whether a dataset is installed.
+
+        Several flavors of "installed" can be tested. By default, a dataset is
+        installed if a worktree and a VCS repository are present. Alternative
+        to "complete" is "vcs".
+
+        Parameters
+        ----------
+        ensure: str
+
+        Returns
+        -------
+        bool
+        """
+        # TODO: Define what exactly to test for, when different flavors are
+        # used.
+        if self.get_path() is not None and self.get_vcs() is not None:
+            return True
+        else:
+            return False
```

Comment:
;)

Was intentional, since there will be additional conditions, depending on
flavor. So, this optimization won't last. But yes, in general you're right,
of course.

2016-03-15 17:13 GMT+01:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/support/dataset.py
> <https://github.com/datalad/datalad/pull/374#discussion_r56191394>:
>
> > +        to "complete" is "vcs".
> > +
> > +        Parameters
> > +        ----------
> > +        ensure: str
> > +
> > +        Returns
> > +        -------
> > +        bool
> > +        """
> > +        # TODO: Define what exactly to test for, when different flavors are
> > +        # used.
> > +        if self.get_path() is not None and self.get_vcs() is not None:
> > +            return True
> > +        else:
> > +            return False
>
> Just could not resist to comment on the pattern... Just return expression
> without if
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub:
> https://github.com/datalad/datalad/pull/374/files#r56191394
>